src/internal_rep.rs

```diff
@@ -23,7 +23,36 @@ const DEFAULT_OBJECT_ROLE: &str = "object_r";
 const DEFAULT_DOMAIN_ROLE: &str = "system_r";
 const DEFAULT_MLS: &str = "s0";
 
-pub type TypeMap = BTreeMap<String, TypeInfo>;
+pub struct TypeMap {
+    types: BTreeMap<String, TypeInfo>,
+    #[allow(dead_code)]
+    aliases: BTreeMap<String, String>,
+}
+
+impl TypeMap {
+    pub fn get(&self, key: &str) -> Option<&TypeInfo> {
+        self.types.get(key)
+    }
+
+    pub fn new() -> Self {
+        TypeMap {
+            types: BTreeMap::new(),
+            aliases: BTreeMap::new(),
+        }
+    }
+
+    pub fn insert(&mut self, key: String, value: TypeInfo) {
+        self.types.insert(key, value);
+    }
+
+    pub fn values(&self) -> std::collections::btree_map::Values<'_, String, TypeInfo> {
+        self.types.values()
+    }
+
+    pub fn iter(&self) -> std::collections::btree_map::Iter<'_, String, TypeInfo> {
```

Comment:
It would be more idiomatic to implement an iterator for `TypeMap`.

Reply:
This is a good idea.  I've implemented something along these lines and pushed it just now.  I tried to actually define the types inside the AliasMap impl block, and then name them just "Iter" etc, but I am getting compiler errors around this message: https://github.com/rust-lang/rust/issues/44265

It sounds like the use of generics in defining a custom type may not be fully supported, but I'm not sure as there is a lot of background reading here.

Any suggestions to make this more idiomatic from what I've just updated to?